awx/main/access.py

```diff
@@ -1325,31 +1325,35 @@ class ExecutionEnvironmentAccess(BaseAccess):
 
     def filtered_queryset(self):
         return ExecutionEnvironment.objects.filter(
-            Q(organization__in=Organization.accessible_pk_qs(self.user, 'execution_environment_admin_role')) |
+            Q(organization__in=Organization.accessible_pk_qs(self.user, 'read_role')) |
             Q(organization__isnull=True)
         ).distinct()
 
-    @check_superuser
     def can_add(self, data):
         if not data:  # So the browseable API will work
             return Organization.accessible_objects(self.user, 'execution_environment_admin_role').exists()
-        return self.check_related('organization', Organization, data)
+        if obj.managed_by_tower:
+            raise PermissionDenied
+        if self.user.is_superuser:
+            return True
+        return self.check_related('organization', Organization, data, mandatory=True,
+                                  role_field='execution_environment_admin_role')
 
-    @check_superuser
     def can_change(self, obj, data):
+        if obj.managed_by_tower:
+            raise PermissionDenied
```

Comment:
We should also check, I think, whether `'managed_by_tower'` is being attempted to be set in the data dict.  Maybe.

Reply:
never mind, we set it to a read-only field :facepalm: